lib/mixpanel.ex

```diff
@@ -16,12 +16,27 @@ defmodule Mixpanel do
     Supervisor.start_link(children, opts)
   end
 
-  def track(event, properties \\ [], timestamp \\ nil) do
+  def track(event, properties, timestamp \\ nil) do
     properties = Mixpanel.Util.add_timestamp(properties, timestamp)
     Mixpanel.Client.track(event, properties)
     :ok
   end
 
+  def track(event) when is_binary(event), do: track(event, [])
+  def track(events) when is_list(events) do
+    Mixpanel.Client.track(events)
+    :ok
+  end
+
+  def engage(event) when is_map(event) do
```

Comment:
Both `engage` fns have the same body, could you collapse them into a single fn, please?